### PR TITLE
contrib/elastic/go-elasticsearch: Add support for elasticsearch v8

### PR DESCRIPTION
**Pull request by xrn**
- Removed v6 from directory name
- Deleted tests for Elasticsearch v6
- Added support for v8

### CODE REVIEW

**Comment by xrn** [comment]
@mrkm4ntr as you worked with https://github.com/DataDog/dd-trace-go/pull/1017 - do you have maybe idea why this test is failing?

**Comment by xrn** [comment]
I was able to solve all test cases but for some reason `TestClientErrorCutoffV8` is still failing. If anyone has any idea I will appreciate.

File: contrib/elastic/go-elasticsearch.v7/elastictrace_test.go [hunk 1]
```diff
@@ -23,8 +23,8 @@ import (
 const debug = false
 
 const (
-	elasticV6URL = "http://127.0.0.1:9202"
-	elasticV7URL = "http://127.0.0.1:9203"
+	elasticV7URL = "http://127.0.0.1:9202"
+	elasticV8URL = "http://127.0.0.1:9203"
```
[the remark this answers is not in the document]
Comment (xrn) [reply contrib/elastic/go-elasticsearch.v7/elastictrace_test.go:24]: i added `xpack.security.enabled: false` flag not required anymore

**Comment by xrn** [comment]
@gbbr what do you think? My goal is when ES8 will be merged I want to add support for OpenSearch 2.0 https://github.com/opensearch-project/OpenSearch

**Review by gbbr — CHANGES_REQUESTED** [review]
Please take a moment to read our contribution guidelines: https://github.com/DataDog/dd-trace-go/blob/main/CONTRIBUTING.md.

For changes, we require an issue to be opened to discuss them. It is not clear to me why we are removing something. Generally we should not. That's a breaking change.

**Comment by xrn** [comment]
@gbbr Issue added

**Comment by xrn** [comment]
@gbbr question: Do you know maybe why pull of image https://www.docker.elastic.co/r/elasticsearch/elasticsearch:8.2.0 is working but there are problems with pull https://www.docker.elastic.co/r/elasticsearch/elasticsearch:8.3.0? Should I upgrade something more?

**Comment by ajgajg1134** [comment]
Hello! Thanks for putting up a PR for this, instead of dropping the v6 tests I think we should leave them and just add the v8 tests as well. (Since we will continue to provide tracing for v6 although it's worth us including a comment that it's a deprecated version). I'll go through a add some additional comments as well :)

**Comment by xrn** [comment]
@ajgajg1134 just do confirm with you that we are on the same page.

- In last PR of this type - https://github.com/DataDog/dd-trace-go/pull/1017 - you removed v5 supprt
- Golang is making legacy versions

But here we would like to leave v6 even  deprecated version and I should just add v8?

**Comment by katiehockman** [comment]
@xrn

> In last PR of this type - https://github.com/DataDog/dd-trace-go/pull/1017 - you removed v5 supprt

The PR description is misleading. We didn’t remove v5 support, we just didn't added new tests for v5 at the time of that PR, since this was the first time we were adding support for elastic search. The PR was saying that it wasn't going to add new tests for something that was already legacy when the support was being introduced.

> For Golang you are making legacy versions as well https://github.com/DataDog/dd-trace-go/pull/1410

Supporting old versions of the language is very different from supporting old versions of a library. We already have a documented policy around language version support. But there’s no need to break backwards compatibility just because something is deprecated, at least not without a migration plan.

> But here we would like to leave v6 even deprecated version and I should just add v8?

Yes. There's no need to delete tests and possibly break backwards compatibility at this time.

**Comment by xrn** [comment]
@katiehockman thanks for response in next week I will update PR

**Comment by xrn** [comment]
@katiehockman made some changes, I think it should be fine now. Not sure why but Kafka test are failing, even if I do not introduced any changes. 

FYI I would be in a trip next 2 weeks if you would like to merge earlier feel free to take over the PR

**Pull request closed without merging**